app/controllers/homepage_controller.rb

```diff
@@ -286,28 +297,31 @@ def search_coordinates(latlng)
     end
   end
 
-  def location_search_params(latlng, distance_unit, distance_max, minimum_distance_max, limit_by_distance)
-    # Current map doesn't react to zoom & panning, so we fetch all the results as before.
-    if @view_type != 'map'
-      Maybe(latlng)
-        .map {
-          distance_limit = [minimum_distance_max, distance_max.to_f].max if limit_by_distance
-          search_coordinates(latlng).merge({
-            distance_unit: distance_unit,
-            distance_max: distance_limit,
-            sort: :distance
-          }).compact
-        }
-        .or_else({})
-    else
-      {}
-    end
-  end
-
   def no_current_user_in_private_clp_enabled_marketplace?
     CustomLandingPage::LandingPageStore.enabled?(@current_community.id) &&
       @current_community.private &&
       !@current_user
   end
 
+  def search_modes_in_use(q, lc, main_search)
+    [
+      lc && (main_search == :location || main_search == :keyword_and_location),
+      q && (main_search == :keyword || main_search == :keyword_and_location)
+    ]
+  end
```

Comment:
Does this return an array of booleans which indicates what search modes are in use? Would it be clearer if the search mode order was not swapped (param order vs. return value)?